controllers/runner_controller.go

```diff
@@ -88,16 +88,20 @@ func (r *RunnerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	} else {
 		finalizers, removed := removeFinalizer(runner.ObjectMeta.Finalizers)
-
+		
 		if removed {
-			ok, err := r.unregisterRunner(ctx, runner.Spec.Organization, runner.Spec.Repository, runner.Name)
-			if err != nil {
-				log.Error(err, "Failed to unregister runner")
-				return ctrl.Result{}, err
-			}
+			if len(runner.Status.Registration.Token) > 0 {
+				ok, err := r.unregisterRunner(ctx, runner.Spec.Organization, runner.Spec.Repository, runner.Name)
+				if err != nil {
+					log.Error(err, "Failed to unregister runner")
+					return ctrl.Result{}, err
```

Comment:
Alternative (or additional) we could even continue on this error - just continue to remove the runner regardless of any failures.

Reply:
Perhaps it can leave the github runner remain registered and never removed when there was a transient error, for example when GitHub API limit is met?